content/en/blog/2022/k8sattributes.md

```diff
@@ -0,0 +1,255 @@
+
+```yaml
+processors:
+  k8sattributes:
+    filter:
+      namespace: my_namespace
+```
+
+In this way you can create a `Role` and don’t need to create a `ClusterRole`,
+reducing the scope of the collector service account to a single namespace.
+
+## Using Resource detector processor
+
+Recently the OpenTelemetry operator incorporated a new feature introduced in
+version v0.50.0, see
+[PR #832](https://github.com/open-telemetry/opentelemetry-operator/pull/832).
+
+This feature sets the `OTEL_RESOURCE_ATTRIBUTES` environment variable on the
+collector container with the k8s pod attributes. It lets you to use the
+resource detector processor, which attaches the environment variable values to
+the spans. This only works when the collector is deployed in sidecar mode.
+
+For example if you deploy this manifest:
+
+```yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: sidecar-for-my-app
+spec:
+  mode: sidecar
+  image: ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector-contrib:0.47.0
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+          thrift_binary:
+          thrift_compact:
+          thrift_http:
+      otlp:
+        protocols:
+          grpc:
+          http:
+
+    processors:
+      resourcedetection:
+        detectors: [env]
+        timeout: 2s
+        override: false
+
+    exporters:
+      jaeger:
+        endpoint: jaeger-all-in-one-collector:14250
+        tls:
+          insecure: true
+
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp, jaeger]
+          processors: [resourcedetection]
+          exporters: [jaeger]
+```
+
+And then deploy the vertx app example, you can see the `OTEL_RESOURCE_ATTRIBUTES`
+environment variable was injected with some values in the sidecar container. Some of them use the Kubernetes downward API to get the attribute values.
+
+An example of the value of the env var:
+
+```yaml
+- name: OTEL_RESOURCE_ATTRIBUTES
+  value: k8s.deployment.name=dep-vertex,k8s.deployment.uid=ef3fe26b-a690-4746-9119-d2dbd94b469f,
+  k8s.namespace.name=default,k8s.node.name=$(OTEL_RESOURCE_ATTRIBUTES_NODE_NAME),k8s.pod.name=
+  (OTEL_RESOURCE_ATTRIBUTES_POD_NAME),k8s.pod.uid=$(OTEL_RESOURCE_ATTRIBUTES_POD_UID),k8s.replicaset
+  name=dep-vertex-59b6f76585,k8s.replicaset.uid=5127bc38-e298-40e1-95df-f4a777e3176c
+```
+
+## Conclusion
+
+We have learned about how to do a basic configuration that allows the
+OpenTelemetry collector to attach Kubernetes resource attributes to traces.
+There could be other complex scenarios but I hope this could serve as a base
+that allows the reader to understand how the to add Kubernetes attributes to the
+spans generated by the pods. We explored how to configure the k8sattributes
+operator in order to accomplish this, and saw some improvements introduced on
+the OpenTelemetry operator that allow us to do this in a simple way on sidecar
+mode
```

Comment:
I struggled to do minor copy-edits on this one, so this is kind of a rewrite of the paragraph, but with some stuff stripped out.

```suggestion
This post covers how to configure the OpenTelemetry collector to attach
Kubernetes resource metadata as resource attributes to OpenTelemetry traces.
The scenarios covered, although basic, illustrate how to add this kind of metadata
to traces so that you can incorporate the technique into other more sophisticated
scenarios.
```

I have half a mind to just suggest dropping the conclusion. I tend not to like conclusions that just summarize what was already covered without an explicit call to action or place to go learn more about a topic. What do you think?

Reply:
I agree, may be I should add a paragraph about how you can learn more about other processor options and different  configurations?